src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -1974,30 +1985,39 @@ GeneratorController::GeneratorController(std::string const &objectName,
         generatorType = GeneratorType::iCEngine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorICEngine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_ICEngine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:CombustionTurbine")) {
         generatorType = GeneratorType::combTurbine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorCombTurbine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_CTurbine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:MicroTurbine")) {
         generatorType = GeneratorType::microturbine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorMicroturbine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_MicroTurbine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:Photovoltaic")) {
         generatorType = GeneratorType::pV;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorPV;
         compPlantTypeOf_Num = DataPlant::TypeOf_PVTSolarCollectorFlatPlate;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:PVWatts")) {
         generatorType = GeneratorType::pvWatts;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorPVWatts;
         compPlantTypeOf_Num = DataPlant::TypeOf_Other;
```

Comment:
Since Generator:Photovoltaic above set compPlantName, why doesn't this object, Generator:PVWatts, need to set compPlantName? Same question for Generator:WindTurbine below. Is compPlantName only used somehow for heat reclaim?

Reply:
those types of generators don't have thermal generation. Only "CHP" types would be connected to plant